### PR TITLE
Print absolute time when watching event

### LINKED ISSUE #13239: Display absolute time for event when using --watch
We now display relative time for event as

    FIRSTSEEN   LASTSEEN
    14m         14m 

But when using `kubectl get ev --watch` the relative time is caculated everytime when the event is printed, in this case for some event occured almost same time, the printed time may be different. 

with wacth:

    18s       18s       1         nginx-controller-9pu4x   Pod  
    19s       19s       1         nginx-controller-nqm00   Pod  
    19s       19s       1         nginx-controller-axvdl   Pod  
    20s       20s       1         nginx-controller-tbrjs   Pod  
    20s       20s       1         nginx-controller-q3go7   Pod  
    21s       21s       1         nginx-controller-7hrjm   Pod 
    21s       21s       1         nginx-controller-s94rd   Pod  
    21s       21s       1         nginx-controller-9pu4x   Pod 
    21s       21s       1         nginx-controller-s94rd   Pod  
    22s       22s       1         nginx-controller-axvdl   Pod 

with get:

    45s         45s        1         nginx-controller-nqm00   Pod 
    45s         45s        1         nginx-controller-kwinh   Pod 
    45s         45s        1         nginx-controller-9pu4x   Pod 
    45s         45s        1         nginx-controller-7hrjm   Pod 
    45s         45s        1         nginx-controller-7hrjm   Pod 
    45s         45s        1         nginx-controller-7hrjm   Pod  
    45s         45s        1         nginx-controller-q2xz9   Pod 
    45s         45s        1         nginx-controller-s94rd   Pod
    45s         45s        1         nginx-controller-s94rd   Pod 
    45s         45s        1         nginx-controller-q3go7   Pod 
    45s         45s        1         nginx-controller-tbrjs   Pod  

with watch it seems these events are not happening at same time, it will give user a misunderstanding and is not helpful.

Should we print the abolute time for event in this case?

@kubernetes/kubectl @jiangyaoguo

### PR DESCRIPTION
**Pull request by jiangyaoguo**
fixed #13239

### CODE REVIEW

**Comment by ncdc** [comment]
cc @kubernetes/rh-cluster-infra @kubernetes/rh-ux

**Comment by liggitt** [comment]
Seems like we should have an options struct at this point, rather than 5 boolean args

File: pkg/kubectl/resource_printer.go [hunk 1]
```diff
@@ -1186,26 +1188,37 @@ func printPersistentVolumeClaim(pvc *api.PersistentVolumeClaim, w io.Writer, wit
...
 	namespace := event.Namespace
 	if withNamespace {
 		if _, err := fmt.Fprintf(w, "%s\t", namespace); err != nil {
 			return err
 		}
 	}
+
+	// While watching event, we should print absolute time.
+	var FirstTimestamp, LastTimestamp string
+	if watch {
```
Comment (liggitt) [inline pkg/kubectl/resource_printer.go]: Name the param based on its effect, like absoluteTimestamps, etc

File: pkg/kubectl/cmd/util/helpers.go [hunk 2]
```diff
@@ -214,6 +214,19 @@ func UsageError(cmd *cobra.Command, format string, args ...interface{}) error {
 	return fmt.Errorf("%s\nSee '%s -h' for help and examples.", msg, cmd.CommandPath())
 }
 
+// Whether this cmd need watching objects.
+func beWatching(cmd *cobra.Command) bool {
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/util/helpers.go]: Nit: call this isWatch

**Comment by jlowdermilk** [comment]
+1 to an options struct for printFunc.

File: pkg/kubectl/resource_printer.go (hunk 1 above)
Comment (jiangyaoguo) [inline pkg/kubectl/resource_printer.go]: These params could be absolute time or relative time. I think "xxTimestamp" is enough.

File: pkg/kubectl/resource_printer.go [hunk 3]
```diff
@@ -1182,30 +1191,41 @@ func printPersistentVolumeClaim(pvc *api.PersistentVolumeClaim, w io.Writer, wit
...
-	if withNamespace {
+	if options.withNamespace {
 		if _, err := fmt.Fprintf(w, "%s\t", namespace); err != nil {
 			return err
 		}
 	}
+
+	// While watching event, we should print absolute time.
+	var FirstTimestamp, LastTimestamp string
+	if options.watch {
```
Comment (jlowdermilk) [inline pkg/kubectl/resource_printer.go]: Would prefer to call this `options.absoluteTimestamps`, as @liggitt suggested

Comment (jiangyaoguo) [inline pkg/kubectl/resource_printer.go]: Sorry for misunderstanding @liggitt 's suggestion. Fixed it.
@jlowdermilk

**Comment by jlowdermilk** [comment]
lgtm. Please squash commits

**Pull request merged**